src/command_modules/azure-cli-lab/azure/cli/command_modules/lab/validators.py

```diff
@@ -0,0 +1,314 @@
+
+logger = azlogging.get_az_logger(__name__)
+devtestlabs_management_client = get_devtestlabs_management_client(None)
+
+
+def get_complex_argument_processor(expanded_arguments, assigned_arg, model_type):
+    """
+    Return a validator which will aggregate multiple arguments to one complex argument.
+    """
+    def _expension_valiator_impl(namespace):
+        """
+        The validator create a argument of a given type from a specific set of arguments from CLI
+        command.
+        :param namespace: The argparse namespace represents the CLI arguments.
+        :return: The argument of specific type.
+        """
+        ns = vars(namespace)
+        kwargs = dict((k, ns[k]) for k in ns if k in set(expanded_arguments))
+
+        setattr(namespace, assigned_arg, model_type(**kwargs))
+
+    return _expension_valiator_impl
+
+
+def validate_lab_vm_create(namespace):
+    """ Validates parameters for lab vm create and updates namespace. """
+    _validate_location(namespace)
+    _validate_expiration_date(namespace)
+    _validate_image_argument(namespace)
+    validate_authentication_type(namespace)
+    _validate_network_parameters(namespace)
+
+
+def validate_lab_vm_list(namespace):
+    """ Validates parameters for lab vm list and updates namespace. """
+    collection = [namespace.filters, namespace.my_vms, namespace.claimable]
+    if not _single(collection):
+        raise CLIError("usage error: [--filters FILTER | --my-vms | --claimable]")
+
+    if namespace.my_vms:
+        # Find out owner object id
+        from azure.cli.core._profile import Profile, CLOUD
+        profile = Profile()
+        cred, _, tenant_id = profile.get_login_credentials(
+            resource=CLOUD.endpoints.active_directory_graph_resource_id)
+        graph_client = GraphRbacManagementClient(cred,
+                                                 tenant_id,
+                                                 base_url=CLOUD.endpoints.active_directory_graph_resource_id)
+        subscription = profile.get_subscription()
+        object_id = namespace.object_id or _get_object_id(graph_client, subscription=subscription)
+        namespace.filters = "Properties/ownerObjectId eq '{}'".format(object_id)
+    elif namespace.claimable:
+        namespace.filters = 'properties/allowClaim'
+    else:
+        namespace.filters = namespace.filters
+
+
+def _validate_location(namespace):
+    """
+    Selects the default location of the lab when location is not provided while creating vm.
+    """
+    if namespace.location is None:
+        lab_operation = devtestlabs_management_client.lab
+        lab = lab_operation.get_resource(namespace.resource_group, namespace.lab_name)
+        namespace.location = lab.location
+
+
+def _validate_expiration_date(namespace):
+    """
+    Validates expiration date if provided.
+    """
+    if namespace.expiration_date:
+        if datetime.datetime.utcnow().date() >= dateutil.parser.parse(namespace.expiration_date).date():
```

Comment:
Generally we prefer to let the server do the validation. The only times we opt for client-side validation are when it would take a long time to hear from the server or if the server/SDK don't do validation. So +1 to @ericame